projects/core/src/modules/scrollable-host/scrollable-host.service.ts

```diff
@@ -14,30 +14,109 @@ export class SkyScrollableHostService {
     private windowRef: SkyAppWindowRef
   ) { }
 
-  public getScrollabeHost(elementRef: ElementRef): HTMLElement | Window {
+  /**
+   * Returns the given element's current scrollable host
+   * @param elementRef The element whose scrollable host is being requested
+   * @returns The current scrollable host
+   */
+  public getScrollableHost(elementRef: ElementRef): HTMLElement | Window {
     return this.findScrollableHost(elementRef.nativeElement);
   }
 
-  public watchScrollableHost(elementRef: ElementRef, completionObservable: Observable<void>): Observable<HTMLElement | Window> {
-    let scrollableHost = this.findScrollableHost(elementRef.nativeElement);
-    let behaviorSubject = new BehaviorSubject(scrollableHost);
-
-    const mutationObserver = this.mutationObserverSvc.create(() => {
-      let newScrollableHost = this.findScrollableHost(elementRef.nativeElement);
-
-      if (newScrollableHost !== scrollableHost) {
-        scrollableHost = newScrollableHost;
+  /**
+   * Returns an observable which emits the given element's current scrollable host
+   * @param elementRef The element whose scrollable host is being requested
+   * @param completionObservable An observable which alerts the internal observers that they should complete
+   * @returns An observable which emits the current scrollable host
+   * @internal
+   */
+  public watchScrollableHost(elementRef: ElementRef): Observable<HTMLElement | Window> {
+    let subCount = 0;
+    let subscribers: Subscriber<HTMLElement | Window>[] = [];
+    let mutationObserver: MutationObserver;
+
+    return new Observable((newSubscriber) => {
+      subCount++;
+
+      let scrollableHost = this.findScrollableHost(elementRef.nativeElement);
+      if (subCount === 1) {
+
+        mutationObserver = this.mutationObserverSvc.create(() => {
+          let newScrollableHost = this.findScrollableHost(elementRef.nativeElement);
+
+          if (newScrollableHost !== scrollableHost) {
+            scrollableHost = newScrollableHost;
+            this.observeForScrollableHostChanges(scrollableHost, mutationObserver);
+
+            for (let subscriber of subscribers) {
+              subscriber.next(scrollableHost);
+            }
+          }
+        });
         this.observeForScrollableHostChanges(scrollableHost, mutationObserver);
-        behaviorSubject.next(scrollableHost);
       }
+      newSubscriber.next(scrollableHost);
+      subscribers.push(newSubscriber);
+
+      newSubscriber.add(() => {
+        subCount--;
+
+        subscribers.filter((subscriber) => { return subscriber !== newSubscriber });
```

Comment:
I'm assuming this is meant to remove an item from the `subscribers` array, but `filter()` returns a new array and doesn't alter the original array. You'll need to reassign `subscribers` if you're going to use `filter()`:

```
subscribers = subscribers.filter((subscriber) => { return subscriber !== newSubscriber });
```

Alternatively you can use `splice()` if you want to modify the original array:

```
const subIndex = subscribers.indexOf(newSubscriber);

if (subIndex >= 0) {
  subscribers.splice(subIndex, 1);
}
```